### PR TITLE
Remove dependency on `dotnet user-secrets init` for secrets initialization

### LINKED ISSUE #21: Initialize user secrets does not work with older `dotnet user-secrets`
I tested the new functionality on my Ubuntu machine, and the initialize action failed without error.

After debugging I discovered the following problems:
1. `dotnet user-secrets --version` (used to check if the tool is installed) returns the following error:
```
Permission denied to modify the '/usr/share/dotnet/sdk/NuGetFallbackFolder' folder.

Here are some options to fix this error:
----------------------------------------
1. Re-run this command with elevated access.
2. Disabled the first run experience by setting the environment variable DOTNET_SKIP_FIRST_TIME_EXPERIENCE to true.
3. Copy the .NET Core SDK to a non-protected location and use it from there.
    
Specify --help for a list of available options and commands.
```
  - This is mitigated by adding `DOTNET_SKIP_FIRST_EXPERIENCE=true` as environment variable to the runtime.
2. The init command returns `Unrecognized command or argument 'init'`
This is due to an older version of the user-secrets tool being installed.
  - This could be solved by checking the version returned by `dotnet user-secrets --version` and adding a "not supported" or "please update" message, when it's older than 3.0.0 (see https://github.com/aspnet/DotNetTools/pull/500)

### PR DESCRIPTION
**Pull request by seclerp**
This PR removes the dependency on `dotnet user-secrets init` command-line call and allows the use of this plugin on .NET Core versions prior to 3.0. 

Now UserSecretsId will be inserted using one of 2 flows:
- **For project files**: `UserSecretsId` node will be added using Rider API for project properties. The same that Rider uses for project properties' UI
- **For shared (`Directory.Build.*`) properties files**: `UserSecretsId` will be added manually using Java XML DOM API. 

The first approach is preferable in all supported cases. The second approach will be used temporarily until the Rider team introduce API for editing `Directory.Buiild.*` files.

Even the second approach works better than `dotnet user-secrets init`, because `init` does not respect formatting (it just adds hardcoded whitespace, see [here](https://github.com/dotnet/aspnetcore/blob/main/src/Tools/dotnet-user-secrets/src/Internal/InitCommand.cs#L123)). I tried to solve the formatting problem as much as Java XML DOM API allowed me to do that.

Basically, there are some formatting drawbacks (e.g. ignoring whitespace before the self-closing tag,  `<Tag />` will be transformed into `<Tag/>` on saving) but such tradeoffs are not critical for a temporary solution.

Closes #21

### CODE REVIEW

**Comment by seclerp** [comment]
@Witik please take a look when you have time

**Comment by seclerp** [comment]
@Witik did you had a chance to look into it?

**Review by Witik — APPROVED** [review]

**Comment by Witik** [comment]
Looks good! I'll test it on my end, and create a release when it works.

**Comment by Witik** [comment]
The test was a success, I've changed the functionality a bit so that it also opens the secrets file in Rider after initializing.

I've submitted version 1.2.0 to Jetbrains for verification, it should be available within a few days (or you can grab the binary from the releases tab)

**Comment by seclerp** [comment]
Brilliant, thanks! It is already available since today's European morning.
